xds/internal/clients/lrsclient/client.go

```diff
@@ -0,0 +1,48 @@
+//revive:disable:unused-parameter
+
+/*
+ *
+ * Copyright 2025 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package lrsclient provides an implementation of the LRS client to report
+// load to the xDS management servers.
+//
+// It allows applications to report load data to an LRS server via the LRS
+// stream. This data can be used for monitoring, traffic management, and other
+// purposes.
+package lrsclient
+
+import (
+	"google.golang.org/grpc/xds/internal/clients"
+)
+
+// LRSClient is an LRS client to report load data to the LRS servers.
+type LRSClient struct {
+}
+
+// ReportLoad starts a load reporting stream to the server in given
+// clients.ServerConfig. All load reports to the same server share the LRS
+// stream.
+//
+// It returns a LoadStore for the user to report loads and a function to
+// cancel the load reporting stream.
+//
+// The stats from LoadStore are reported periodically until cleanup
+// function is called.
+func (c *LRSClient) ReportLoad(serverConfig clients.ServerConfig) (*LoadStore, func()) {
```

Comment:
```suggestion
// ReportLoad creates a new load reporting stream for the client.
func (c *LRSClient) ReportLoad(serverConfig clients.ServerConfig) (*LoadStore) {
```

Could we have a `Stop` method on the `LoadStore` instead of having a separate cancel function?  The separate cancel function is useful for a few specific things, but I don't think there's any need for it here?  @easwars

Reply:
Added a close() method similar to xDS client.

Reply:
I think we initially had a close method on the xDS client type and we eventually changed it to a cancel function, since it was easier to use a onceFunc to cancel the watch or the load report (because we want these cancels to be idempotent). Also, because we return an interface type in the existing implementation where there is two levels of wrapping to provide reference counting. With the new implementation though, we would be returning a concrete type. So, this concrete type could be a wrapper around an unexported real implementation that is reference counted.